src/main/scala/edu/ie3/mobsim/utils/PathsAndSources.scala

```diff
@@ -199,10 +181,15 @@ object PathsAndSources extends LazyLogging {
     */
   private def determineRecentOutputDirectory(baseOutputDir: String): String = {
     /* find last modified directory in output path -> this is where SIMONA writes into during the simulation */
-    val dir = Paths
-      .get(baseOutputDir)
-      .toFile
+    val outputDir = new File(baseOutputDir)
+    val outputPathFiles = outputDir
       .listFiles(DIRECTORY.asInstanceOf[FileFilter])
+
+    if (!outputDir.exists() || outputPathFiles.isEmpty) {
+      return Seq(baseOutputDir, "mobilitySimulator").mkString(File.separator)
+    }
```

Comment:
I think this can be integrated nicely into the functional code below with a `flatMap`. hmu and I'll try to explain